File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.js
function solution(land) {
    for (let i = 1; i < land.length; i++) {
        for (let j = 0; j < 4; j++) {
            // 같은 인덱스는 제외한 이전 land 배열
            const prevLand = land[i - 1].filter((_, _i) => j !== _i)
            land[i][j] += Math.max(...prevLand)
        }
    }
    console.log(land)
    return Math.max(...land.pop())
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
# def solution(land):
#     answer = 0
#     last_col = 0
    
#     for row in land:
#         max_score = 0
        
#         for i in range(len(row)):
#             cur = row[i]
#             if cur > max_score and i != last_col:
#                 max_score = cur
            
#         answer += max_score
#         last_col = row.index(max_score)
#         print(max_score)

#     return answer

def solution(land):
    for i in range(len(land) - 1):
        cur_row = land[i]
        max_score = max(cur_row)
        max_index = cur_row.index(max_score)
        second_max_score = -1
        
        for j in range(len(cur_row)):
            if j != max_index and cur_row[j] > second_max_score:
                second_max_score = cur_row[j]
        
        next_row = land[i + 1]
        for j in range(len(next_row)):
            if j == max_index:
                next_row[j] += second_max_score
            else:
                next_row[j] += max_score
   
    return max(land[-1])